### ISSUE TITLE
Missing prop validation when propTypes is static property

### ISSUE THREAD

**Issue by jprochazk** [yannickcr/eslint-plugin-react#3103]
```jsx
class Test extends React.Component {
  static propTypes = {
    value: PropTypes.string
  };

  render() {
    return <span>{this.props.value}</span>; // 'value' is missing in props validation
  }
}
```

It stopped working after switching to `@typescript-eslint/parser`.

This still works:
```jsx
class Test extends React.Component {
  render() {
    return <span>{this.props.value}</span>;
  }
}

Test.propTypes = {
  value: PropTypes.string
};
```

eslint config:
```json
{
  "extends": [
    "eslint:recommended",
    "plugin:react/recommended",
    "plugin:@typescript-eslint/recommended",
  ],
  "parser": "@typescript-eslint/parser",
  "plugins": ["react", "@typescript-eslint"],
}
```

**Comment by ljharb** [comment]
@eps1lon the latest version of eslint-plugin-react tests against v5 of the TS eslint parser, and tests are passing, so I think you should be good to go!

Happy to reopen if there's still issues.
